src/librustc_trans/trans/_match.rs

```diff
@@ -272,7 +272,10 @@ impl<'a, 'tcx> Opt<'a, 'tcx> {
         }
     }
 
-    fn trans<'blk>(&self, mut bcx: Block<'blk, 'tcx>) -> OptResult<'blk, 'tcx> {
+    fn trans<'r, 'blk>(&self,
+                       &mut Block { bl, ref mut fcx }: &mut Block<'r, 'blk, 'tcx>)
+                       -> OptResult<'blk> {
+        let mut bcx = &mut bl.with(fcx);
```

Comment:
This pattern (well, similar) was actually already used all over the place prior to this PR.  But I'll take a closer look at this particular one.